apps/ledger-live-mobile/e2e/bridge/client.ts

```diff
@@ -22,20 +25,60 @@ export function init(port = 8099) {
   ws.onmessage = onMessage;
 }
 
-async function onMessage(event: { data: unknown }) {
+export const e2eDeviceDescriptorSubject = new Subject<SubjectData>();
+
+type SubjectData =
+  | {
+      type: DescriptorEventType;
+      payload: { id: string; name: string; serviceUUID: string };
+    }
+  | { type: "open" };
+
+export type MessageData =
+  | SubjectData
+  | { type: "acceptTerms" }
+  | { type: "navigate"; payload: string }
+  | { type: "importSettings"; payload: Partial<SettingsState> }
+  | {
+      type: "importAccounts";
+      payload: {
+        data: AccountRaw;
+        version: number;
+      }[];
+    }
+  | {
+      type: "setGlobals";
+      payload: { [key: string]: unknown };
+    }
+  | {
+      type: "mockDeviceAction";
+      payload: { deviceActionToMock: UpdateDeviceActionStateMessage["deviceAction"] };
+    }
+  | {
+      type: "newDeviceActionSate";
+      payload: UpdateDeviceActionStateMessage
+  };
+
+async function onMessage(event: WebSocketMessageEvent) {
   invariant(
     typeof event.data === "string",
     "[E2E Bridge Client]: Message data must be string",
   );
-  const msg = JSON.parse(event.data);
+  const msg: MessageData = JSON.parse(event.data);
   invariant(msg.type, "[E2E Bridge Client]: type is missing");
 
   log(`Message\n${JSON.stringify(msg, null, 2)}`);
 
   switch (msg.type) {
+    case "mockDeviceAction":
+      mockDeviceActions(msg.payload.deviceActionToMock);
+      break;
+    case "newDeviceActionSate":
+      updateDeviceActionState(msg.payload);
+      break;
     case "add":
     case "open":
-      e2eBridgeSubject.next(msg);
+      e2eDeviceDescriptorSubject.next(msg);
```

Comment:
Probably best not to call it specifically `DeviceDescriptor` as this bridge is also used to do other stuff like load user data (the `setGlobals` one I believe)